bench/query.go

```diff
@@ -132,6 +132,90 @@ func (q *QueryGenerator) Random(maxN, depth, maxargs int, idmin, idmax uint64) *
 	}
 }
 
+// RandomRangeQuery returns a randomly generated either sum/range query.
+func (q *QueryGenerator) RandomRangeQuery(depth, maxargs int, frame, field string, idmin, idmax uint64) *pql.Call {
+	val := q.R.Intn(5)
+	switch val {
+	case 1:
+		return q.RandomSum(depth, maxargs, frame, field, idmin, idmax)
+	default:
+		return q.RandomRange(maxargs, field, idmin, idmax)
+	}
+}
+
+func (q *QueryGenerator) RandomRange(numArg int, field string, idmin, idmax uint64) *pql.Call {
+	call := q.R.Intn(4)
+	if call == 0 {
+		return q.RangeCall(field, idmin, idmax)
+	}
+	calls := make([]*pql.Call, numArg)
+	for i := 0; i < numArg; i++ {
+		calls[i] = q.RangeCall(field, idmin, idmax)
+	}
+
+	switch call {
+	case 1:
+		return Difference(calls...)
+	case 2:
+		return Intersect(calls...)
+	case 3:
+		return Union(calls...)
+	}
+
+	return nil
+}
+
+func (q *QueryGenerator) RangeCall(field string, idmin, idmax uint64) *pql.Call {
+
+	var operations = []pql.Token{pql.GT, pql.LT, pql.GTE, pql.LTE}
```

Comment:
should we add `pql.EQ` here?

Reply:
and possibly `pql.BETWEEN`? (although that one has a different argument type, so it may not be as easy to implement here).